### ISSUE TITLE
[Question] How can I use beast with HTTP proxy?

### ISSUE THREAD

**Issue by zjhmale** [boostorg/beast#1216]
Hi, I want to create a beast WebSocket client with my local HTTP proxy, is there any documentation or verbose example to demonstrate how can I achieve that?

**Comment by vinniefalco** [comment]
It is possible, but there are no examples. If you can paste some of the documentation for your proxy explaining what it expects in the initial handshake, I might be able to help.

**Comment by zjhmale** [comment]
OK, currently my old system is written in 100% pure Python code, and we used the [python WebSocket library](https://github.com/websocket-client/websocket-client), I can set up HTTP proxy with that lib easily in [this way](https://github.com/websocket-client/websocket-client#http-proxy), so can I do this like a charm with beast? Thanks in advance :)

**Comment by vinniefalco** [comment]
> can I do this like a charm with beast

No, you will have to implement proxy support yourself. Beast is a low-level library, it only handles the serialization and parsing of the HTTP message.

**Comment by zjhmale** [comment]
@vinniefalco OK, gotcha, thanks :)

**Comment by allen2015555** [comment]
i want to create a wss client with beast and i want to use Fiddler to check the wss client is correct.i like the beast but it's regrettable it doesn't support proxy.

**Comment by vinniefalco** [comment]
You can always implement proxy support yourself, use the HTTP interface to connect to the proxy first.

**Comment by zjhmale** [comment]
@vinniefalco right now I want to implement this HTTP proxy feature, the right start point is to delegate the proxy options when we build HTTP requests here https://github.com/boostorg/beast/blob/0b68ed651b6bc7b681cf440ed6a220089e21473f/include/boost/beast/websocket/impl/stream_impl.hpp#L569-L595 which will be used by WebSocket client right?

**Comment by vinniefalco** [comment]
If you want to modify the Upgrade request, then set a decorator on the stream:
https://www.boost.org/doc/libs/1_71_0/libs/beast/doc/html/beast/using_websocket/decorator.html

**Comment by zjhmale** [comment]
make sense, the decorator is more flexible :)
